### PR TITLE
feat(button): add href for Link Variant

### PR DESCRIPTION
**Pull request by tejas161**
## What I did

1. Created new attribute **href** for pf-button Link.
2. Created new attribute **target** for pf-button Link.
3. Updated docs for pf-button Link showing examples how to pass an href with target = "_blank" or target = "_self".

## Testing Instructions

1. Check if pf-button "link" variant properly redirects to the link passed via href attribute.
2. Check the Accessibility of pf-button Link.

Closes: #2769

### CODE REVIEW

File: .changeset/popular-poets-march.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@patternfly/elements": major
+---
+
+Added href attribute to pf-button Link variant
```
Comment (bennypowers) [inline .changeset/popular-poets-march.md:5]:
```suggestion
Added `href` attribute to `<pf-button variant="link">`
```

**Review by bennypowers — APPROVED** [review]
@adamjohnson can you take a look at this for usability?

File: elements/pf-button/pf-button.ts [hunk 2]
```diff
@@ -214,18 +220,39 @@ export class PfButton extends LitElement {
...
     this.addEventListener('click', this.#onClick);
     this.addEventListener('keydown', this.#onKeydown);
-    this.tabIndex = 0;
   }
 
   protected override willUpdate(): void {
     this.#internals.ariaLabel = this.label || null;
     this.#internals.ariaDisabled = String(!!this.disabled);
+    if (this.variant !== 'link' || (this.variant === 'link' && !this.href ) ) {
+      this.tabIndex = 0;
```
Comment (bennypowers) [inline elements/pf-button/pf-button.ts:229]:
```suggestion
      this.tabIndex = 0;
      this.#internals.role = 'button';
```

File: elements/pf-button/pf-button.ts [hunk 3]
```diff
@@ -214,18 +220,39 @@ export class PfButton extends LitElement {
...
+    }
```
Comment (bennypowers) [inline elements/pf-button/pf-button.ts:230]:
```suggestion
    } else {
      this.tabIndex = -1;
      this.#internals.role = 'none';
    }
```

**Review by zeroedin — COMMENTED** [review]
@bennypowers seems like you can click around the href variant and not actually follow the link.  The click box for the `<a>` doesn't appear to cover the whole element clickable area.

**Comment by adamjohnson** [comment]
If you try to tab to "Link" and "Go to PatternFly" in the [Link Variant](https://deploy-preview-2814--patternfly-elements.netlify.app/components/button/#link-variant) demo, they get skipped and are not focusable.

It looks like removing `tabindex="-1"` from the host resolves this issue.

**Pull request merged**